### PR TITLE
Using WKB as GeometrySerializer implementation

### PR DESCRIPTION
**Pull request by netanel246**
## Is this PR related to a proposed Issue?
[#424](https://github.com/DataSystemsLab/GeoSpark/issues/424)
## What changes were proposed in this PR?
Replaced GeometrySerializer to use WKB API instead of the ShapeSerde which contains bugs (added test case with a buggy multipolygon)
## How was this patch tested?
Added test for the buggy multipolygon case
## Did this PR include necessary documentation updates?
Yes

### CODE REVIEW

**Comment by jiayuasu** [comment]
@netanel246 It looks like the CI failed at the serialization in Python.

@Imbruced Hi Pawel, did you write any specific Python wrapper for GeoSpark serializer and deserializer? We were trying to fix a bug in GeoSpark Java/Scala serializer...

**Comment by Imbruced** [comment]
@jiayuasu I am using scala/Java serializer to create shapely objects. I will look how to integrate your changes with Python API today.

**Comment by Imbruced** [comment]
@jiayuasu Finally I had chance to look at that. So you decide to drop previous serializer and use wkb as a binary representation of geometry. That will require some changes in Python and scala code which is used in wrapper. Maybe this is good time to merge my Scala code to main repo ?

**Comment by jiayuasu** [comment]
@Imbruced 

Sure, push the Scala code to the main repo. But I thought you have pushed your wrapper to the "python" folder, right?

**Comment by jiayuasu** [comment]
@Imbruced BTW, there is another question in GeoSpark Gitter chat that asks for an issue in Python wrapper.

**Comment by Imbruced** [comment]
@jiayuasu Is this going for release 1.3.2 ? From my site it require few changes. Did you compare serialization time between old approach and current one ? From my site I will use shapely.wkb loads and dumps to use the same approach. I hope there are not differences between shapely and jts implementation (wkb is a standard).

**Comment by netanel246** [comment]
@jiayuasu , @Imbruced 
Is there anything I can do to promote it?

**Comment by Imbruced** [comment]
Python API needs huge change and I think this deserve 1.3.2 version. All translation between jts geometries and shapely geometries was based on GeometrySerializer instance.

**Comment by jiayuasu** [comment]
@Imbruced Thanks, Pawel.

Yes, we can do it in 1.3.2. But why do you think this requires a huge change?

I believe the API remains the same, right?

**Comment by Imbruced** [comment]
@jiayuasu I mean it is not 1 hour of work but couple of hours. From user point of view, API does not change, but serialization may lead to performance change. I will do some benchmarks.

**Comment by netanel246** [comment]
There are some updates on that?
There is something I can do to make it happen?

**Comment by Imbruced** [comment]
@netanel246 It is almost there. ~5% out of 400 tests does not pass so it should be ready during this week. You can check the code here https://github.com/Imbruced/GeoSpark/tree/wkb-serializer. I will rebase once it finished and create PR to your branch and then we will merge with master.

**Comment by netanel246** [comment]
Cool. Thank you!!

**Comment by Imbruced** [comment]
@netanel246 
All 400 test cases pass. I have to make sure that everything works fine and do some benchmarks to compare previous serialization with current one. As I said, it should be ready during this week.

**Comment by netanel246** [comment]
The build failed because of the geotools repo is not accessible,
I tried the two possible repositories from the [maven repo](https://mvnrepository.com/artifact/org.geotools/gt-main/17.0): 

1. [Boundless](https://repo.boundlessgeo.com/main/)
2. [OSGeo Old](https://download.osgeo.org/webdav/geotools/)

What can we do about it? Do you know someone in geotools team who can handle that?

**Comment by jiayuasu** [comment]
@netanel246 That is strange. GeoSpark has Geotools as a dependency before. It worked. Did you update the GeoTools version?

**Comment by netanel246** [comment]
@jiayuasu , no we don't update the dependencies, only the version of Geospark

**Comment by Imbruced** [comment]
@netanel246 
I think you can remove that test (test serializing with user Data). I forget to commit an update on it.

**Comment by netanel246** [comment]
@Imbruced , I fixed the dependency issue but now it's failed on the `test serializing with user Data`.
I try it on my host and the test keeps failing.
Maybe the `expectedEncodedArray` should be different?

**Comment by Imbruced** [comment]
@netanel246 I added comment :) Yeah, the output should be different. For now you can remove it, other tests handle use cases.

**Comment by netanel246** [comment]
@Imbruced, I saw that you've created `testWkb` function in **the org.datasyslab.geosparksql** package. Can you move it to the test package? or can I remove it?
Also, `Passed St_GeomFromWKT` in **org.datasyslab.geosparksql** are fail to run.
Can I remove it too?

**Comment by Imbruced** [comment]
@netanel246 please do that. Sorry if I left some additional files.

**Comment by netanel246** [comment]
@Imbruced , somehow the following command not succeeded:
`find python/geo_wrapper/target/scala-2.11 -iregex "python\/geo_wrapper\/target\/scala-2\.11\/geo_wrapper_2\.11-[0-9]\.[0-9]\.[0-9]\.jar" -exec cp {} $SPARK_HOME/jars \;`

**Comment by Imbruced** [comment]
@netanel246 I dont know od merge went wrong, but this is no longer needed. That code od placed now in core od sql GeoSpark package.

**Comment by netanel246** [comment]
@Imbruced, can you please share with me how the travis.yml should look like?
I cannot find it...

**Comment by Imbruced** [comment]
@netanel246 I removed old dependencies from script, that build should be fine.

**Comment by Imbruced** [comment]
@netanel246 I will assign which files should be removed (in code review), Can you remove them after that ?

**Comment by Imbruced** [comment]
Or I will do that and I will go through the code again and make sure that we do not have additional things.

File: python/Pipfile [hunk 1]
```diff
@@ -21,6 +21,7 @@ pandas="*"
 geopandas="==0.6.0"
 pyspark="*"
 attrs="*"
+twine = "*"
```
Comment (Imbruced) [inline python/Pipfile:24]: Remove unused libs like twine, leave only essential ones.

File: python/Pipfile.lock [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
```
Comment (Imbruced) [inline python/Pipfile.lock:1]: Update Pipfile.lock

**Comment by jiayuasu** [comment]
Is this PR ready to be merged?

**Comment by netanel246** [comment]
@jiayuasu , I think it does.

**Comment by jiayuasu** [comment]
@netanel246 In this PR, you didn't touch anything for geometry serialization in RDD API, right? It looks like you only add the WKB serializer to SQL API?

**Comment by netanel246** [comment]
@jiayuasu yep. I didn't change the RDD API.

**Pull request merged**